### DISCUSSION TITLE
Horizontal expanding selection

### DISCUSSION THREAD

**Discussion by OleksandrTaranFrienton** [nick-keller/react-datasheet-grid#266] [Feature requests]
Hi, thank you at first for this awesome library.
The docs tell about the "expanding selection" feature that works only vertically, i.e. values are copied from a row to the next rows.
A common Excel-like function is to have horizontal expansion that would be the same as https://react-datasheet-grid.netlify.app/docs/features#expanding-selection but instead copying values from a column to the next columns.
Are you planning to support it?

Thank you in advance!

**Comment by nick-keller** [comment]
Hi @OleksandrTaranFrienton and thx for the feedback ;)
This feature is not planned because it does not feel right. DSG is column based, it makes sense to expand a selection vertically because a column has all its rows similar. But horizontally? What if you have a number, a country, and a date column, shat should happen?

I hope you see my point, if you have a counter argument feel free to explain and I will consider it, in the meantime I'm solving this discussion ;)

**Reply by OleksandrTaranFrienton** [reply]
Hi @nick-keller . Thank you for a fast reply. I understand your point if one thinks as Airtable use case. The use case I'm facing is a "standard financial planning or spreadsheet". Columns are months and rows are different kind of income and expenses.
In such a planning scenario some sequential cells in the same row have the same data type (string or number). I want to provide the value for the first month and then drag its value to the next cells in this row.

So it'd be very applicable in case multiple horizontally sequential cells have the same data type.

PS: maybe this lib is not the best choice for such a use case...

**Reply by fmal** [reply]
@nick-keller curious why the expand only works for rows below active cell within column. Would it be possible to make to expand  upwards as well?
